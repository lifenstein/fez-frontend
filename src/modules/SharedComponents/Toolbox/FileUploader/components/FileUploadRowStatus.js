import React, { PureComponent, Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Tooltip from '@material-ui/core/Tooltip';
import CircularProgress from '@material-ui/core/CircularProgress';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';

import Delete from '@material-ui/icons/Delete';
import Done from '@material-ui/icons/Done';

export class FileUploadRowStatus extends PureComponent {
    static propTypes = {
        progress: PropTypes.number,
        isUploadInProgress: PropTypes.bool,
        disabled: PropTypes.bool,
        onDelete: PropTypes.func.isRequired,
        locale: PropTypes.object,
        name: PropTypes.string,
        fileUploadRowStatusId: PropTypes.string,
    };

    static defaultProps = {
        locale: {
            deleteHint: 'Remove this file',
            uploadInProgressText: 'Uploading...',
        },
    };

    render() {
        const { isUploadInProgress, disabled, progress } = this.props;
        const { deleteHint, uploadInProgressText } = this.props.locale;
        const progressProps = progress > 0 ? { variant: 'determinate', value: progress } : {};

        return (
            <Fragment>
                {!isUploadInProgress && (
                    <Tooltip title={deleteHint}>
                        <span>
                            <IconButton
                                onClick={this.props.onDelete}
                                disabled={disabled}
                                data-testid={`${this.props.fileUploadRowStatusId}-delete`}
                            >
                                <Delete />
                            </IconButton>
                        </span>
                    </Tooltip>
                )}
                {isUploadInProgress && progress !== 100 && (
                    <Fragment>
                        <CircularProgress {...progressProps} size={20} thickness={4} />
                        <Typography variant="caption" aria-label={progress > 0 ? `${progress}%` : uploadInProgressText}>
                            {progress > 0 ? `${progress}%` : uploadInProgressText}
                        </Typography>
                    </Fragment>
                )}
                {isUploadInProgress && progress === 100 && (
                    <IconButton>
                        <Done />
                    </IconButton>
                )}
            </Fragment>
        );
    }
}

export const mapStateToProps = (state, ownProps) => {
    return {
        progress: (!!state.get('fileUpload') && state.get('fileUpload')[ownProps.name]) || 0,
        isUploadInProgress: !!state.get('fileUpload') && state.get('fileUpload').isUploadInProgress,
    };
};

export default connect(mapStateToProps)(FileUploadRowStatus);
