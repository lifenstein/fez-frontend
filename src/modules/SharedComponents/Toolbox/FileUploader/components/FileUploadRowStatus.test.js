import React from 'react';
import { FileUploadRowStatus, mapStateToProps } from './FileUploadRowStatus';
import { render, WithReduxStore } from 'test-utils';

const getProps = (testProps = {}) => ({
    progress: 0,
    isUploadInProgress: false,
    onDelete: jest.fn(),
    ...testProps,
});

function setup(testProps = {}) {
    return render(
        <WithReduxStore>
            <FileUploadRowStatus {...getProps(testProps)} />
        </WithReduxStore>,
    );
}

describe('Component FileUploadRowStatus', () => {
    it('should render default view', () => {
        const { container } = setup();
        expect(container).toMatchSnapshot();
    });

    it('should render circular progress if upload is in progress', () => {
        const { container } = setup({
            progress: 50,
            isUploadInProgress: true,
        });
        expect(container).toMatchSnapshot();
    });

    it('should render done icon if upload is finished', () => {
        const { container } = setup({
            progress: 100,
            isUploadInProgress: true,
        });
        expect(container).toMatchSnapshot();
    });

    it('should render disabled view', () => {
        const { container } = setup({ disabled: true });
        expect(container).toMatchSnapshot();
    });

    it('should render for edge browser if file is being uploaded but no progress data', () => {
        const { container } = setup({ progress: 0, isUploadInProgress: true });
        expect(container).toMatchSnapshot();
    });

    it('should render if file uploaded successfully but later other file failed', () => {
        const { container } = setup({ progress: 100, isUploadInProgress: false });
        expect(container).toMatchSnapshot();
    });

    it('should map state to props as expected', () => {
        const state = {
            get: jest.fn(() => ({
                propName: 'test1',
                isUploadInProgress: true,
            })),
        };
        const test = mapStateToProps(state, { name: 'propName' });
        const result = {
            progress: 'test1',
            isUploadInProgress: true,
        };
        expect(test).toMatchObject(result);

        const test2 = mapStateToProps(
            {
                get: jest.fn(() => null),
            },
            { name: 'propName' },
        );
        const result2 = {
            progress: 0,
            isUploadInProgress: false,
        };
        expect(test2).toMatchObject(result2);
    });
});
