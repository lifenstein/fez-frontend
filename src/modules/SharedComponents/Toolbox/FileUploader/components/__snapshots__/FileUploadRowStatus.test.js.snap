// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component FileUploadRowStatus should mount the container 1`] = `
<Provider
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Immutable.Map {},
          },
          Object {
            "type": "return",
            "value": Immutable.Map {},
          },
          Object {
            "type": "return",
            "value": Immutable.Map {},
          },
          Object {
            "type": "return",
            "value": Immutable.Map {},
          },
        ],
      },
      "subscribe": [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
          Array [
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      },
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "key": "testKey",
          "pathname": "/",
        },
      ]
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <ThemeProvider
        theme={
          Object {
            "breakpoints": Object {
              "between": [Function],
              "down": [Function],
              "keys": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
              "only": [Function],
              "up": [Function],
              "values": Object {
                "lg": 1280,
                "md": 960,
                "sm": 600,
                "xl": 1920,
                "xs": 0,
              },
              "width": [Function],
            },
            "direction": "ltr",
            "graphs": Object {
              "color1": "#d72865",
              "color2": "#1e7d94",
              "color3": "#49075E",
              "color4": "#468fcc",
              "color5": "#f28620",
              "color6": "#333333",
              "color7": "#595959",
            },
            "hexToRGBA": [Function],
            "mixins": Object {
              "gutters": [Function],
              "toolbar": Object {
                "@media (min-width:0px) and (orientation: landscape)": Object {
                  "minHeight": 48,
                },
                "@media (min-width:600px)": Object {
                  "minHeight": 64,
                },
                "minHeight": 56,
              },
            },
            "overrides": Object {
              "MUIDataTable": Object {
                "paper": Object {
                  "boxShadow": "none",
                  "margin": 0,
                  "padding": 0,
                },
              },
              "MuiFormLabel": Object {
                "root": Object {
                  "overflow": "hidden",
                  "textOverflow": "ellipsis",
                  "whiteSpace": "nowrap",
                  "width": "100%",
                },
              },
              "MuiIconButton": Object {
                "colorInherit": Object {
                  "color": "none",
                },
              },
              "MuiMenuItem": Object {
                "root": Object {
                  "&$selected": Object {
                    "backgroundColor": "#4085C6 !important",
                    "color": "#FFFFFF",
                  },
                },
              },
            },
            "palette": Object {
              "accent": Object {
                "dark": "#2a557d",
                "light": "#54acff",
                "main": "#316799",
              },
              "action": Object {
                "activatedOpacity": 0.12,
                "active": "rgba(0, 0, 0, 0.54)",
                "disabled": "rgba(0, 0, 0, 0.26)",
                "disabledBackground": "rgba(0, 0, 0, 0.12)",
                "disabledOpacity": 0.38,
                "focus": "rgba(0, 0, 0, 0.12)",
                "focusOpacity": 0.12,
                "hover": "rgba(0, 0, 0, 0.04)",
                "hoverOpacity": 0.04,
                "selected": "rgba(0, 0, 0, 0.08)",
                "selectedOpacity": 0.08,
              },
              "augmentColor": [Function],
              "background": Object {
                "default": "#fafafa",
                "paper": "#fff",
              },
              "common": Object {
                "black": "#000",
                "white": "#fff",
              },
              "contrastThreshold": 3,
              "divider": "rgba(0, 0, 0, 0.12)",
              "error": Object {
                "contrastText": "#fff",
                "dark": "#790000",
                "light": "#ff0000",
                "main": "#c80000",
              },
              "getContrastText": [Function],
              "grey": Object {
                "100": "#f5f5f5",
                "200": "#eeeeee",
                "300": "#e0e0e0",
                "400": "#bdbdbd",
                "50": "#fafafa",
                "500": "#9e9e9e",
                "600": "#757575",
                "700": "#616161",
                "800": "#424242",
                "900": "#212121",
                "A100": "#d5d5d5",
                "A200": "#aaaaaa",
                "A400": "#303030",
                "A700": "#616161",
              },
              "info": Object {
                "contrastText": "#fff",
                "dark": "#1976d2",
                "light": "#64b5f6",
                "main": "#2196f3",
              },
              "primary": Object {
                "contrastText": "#fff",
                "dark": "#3b1a59",
                "gradient": Object {
                  "diagonal": Object {
                    "background": "linear-gradient(135deg, rgb(81,36,122) 55%,rgb(150,42,139) 100%)",
                    "filter": "progid:DXImageTransform.Microsoft.gradient( startColorstr=\\"#51247a\\", endColorstr=\\"#962a8b\\",GradientType=1 )",
                  },
                  "horizontal": Object {
                    "background": "linear-gradient(to right, rgb(81,36,122) 55%,rgb(150,42,139) 100%)",
                    "filter": "progid:DXImageTransform.Microsoft.gradient( startColorstr=\\"#51247a\\", endColorstr=\\"#962a8b\\",GradientType=1 )",
                  },
                },
                "light": "#962A8B",
                "main": "#51247A",
              },
              "secondary": Object {
                "contrastText": "#fff",
                "dark": "#333333",
                "light": "#f2f2f2",
                "main": "#595959",
              },
              "success": Object {
                "contrastText": "#fff",
                "dark": "#005000",
                "light": "#00a700",
                "main": "#007200",
              },
              "text": Object {
                "disabled": "rgba(0, 0, 0, 0.38)",
                "hint": "rgba(0, 0, 0, 0.38)",
                "primary": "rgba(0, 0, 0, 0.87)",
                "secondary": "rgba(0, 0, 0, 0.54)",
              },
              "tonalOffset": 0.2,
              "type": "light",
              "warning": Object {
                "contrastText": "#fff",
                "dark": "#542400",
                "light": "#ff9a57",
                "main": "#bf5000",
              },
              "white": Object {
                "main": "#FFFFFF",
              },
            },
            "props": Object {},
            "shadows": Array [
              "none",
              "0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)",
              "0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)",
              "0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)",
              "0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)",
              "0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)",
              "0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)",
              "0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)",
              "0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)",
              "0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)",
              "0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)",
              "0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)",
              "0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)",
              "0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)",
              "0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)",
              "0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)",
              "0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)",
              "0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)",
              "0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)",
              "0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)",
              "0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)",
              "0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)",
              "0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)",
              "0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)",
              "0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)",
            ],
            "shape": Object {
              "borderRadius": 4,
            },
            "spacing": [Function],
            "status": Object {
              "danger": "#f44336",
              "warning": "#ff9800",
            },
            "transitions": Object {
              "create": [Function],
              "duration": Object {
                "complex": 375,
                "enteringScreen": 225,
                "leavingScreen": 195,
                "short": 250,
                "shorter": 200,
                "shortest": 150,
                "standard": 300,
              },
              "easing": Object {
                "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
                "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
                "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
                "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
              },
              "getAutoHeightDuration": [Function],
            },
            "typography": Object {
              "body1": Object {
                "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                "fontSize": "1rem",
                "fontWeight": 300,
                "letterSpacing": "0.00938em",
                "lineHeight": 1.5,
              },
              "body2": Object {
                "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                "fontSize": "0.875rem",
                "fontWeight": 300,
                "letterSpacing": "0.01071em",
                "lineHeight": 1.43,
              },
              "button": Object {
                "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                "fontSize": "0.875rem",
                "fontWeight": 400,
                "letterSpacing": "0.02857em",
                "lineHeight": 1.75,
                "textTransform": "uppercase",
              },
              "caption": Object {
                "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                "fontSize": "0.75rem",
                "fontWeight": 300,
                "letterSpacing": "0.03333em",
                "lineHeight": 1.66,
              },
              "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
              "fontSize": 14,
              "fontWeightBold": 700,
              "fontWeightLight": 200,
              "fontWeightMedium": 400,
              "fontWeightRegular": 300,
              "h1": Object {
                "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                "fontSize": "6rem",
                "fontWeight": 200,
                "letterSpacing": "-0.01562em",
                "lineHeight": 1.167,
              },
              "h2": Object {
                "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                "fontSize": "3.75rem",
                "fontWeight": 200,
                "letterSpacing": "-0.00833em",
                "lineHeight": 1.2,
              },
              "h3": Object {
                "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                "fontSize": "3rem",
                "fontWeight": 300,
                "letterSpacing": "0em",
                "lineHeight": 1.167,
              },
              "h4": Object {
                "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                "fontSize": "2.125rem",
                "fontWeight": 300,
                "letterSpacing": "0.00735em",
                "lineHeight": 1.235,
              },
              "h5": Object {
                "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                "fontSize": "1.5rem",
                "fontWeight": 300,
                "letterSpacing": "0em",
                "lineHeight": 1.334,
              },
              "h6": Object {
                "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                "fontSize": "1.25rem",
                "fontWeight": 400,
                "letterSpacing": "0.0075em",
                "lineHeight": 1.6,
              },
              "htmlFontSize": 16,
              "overline": Object {
                "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                "fontSize": "0.75rem",
                "fontWeight": 300,
                "letterSpacing": "0.08333em",
                "lineHeight": 2.66,
                "textTransform": "uppercase",
              },
              "pxToRem": [Function],
              "round": [Function],
              "subtitle1": Object {
                "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                "fontSize": "1rem",
                "fontWeight": 300,
                "letterSpacing": "0.00938em",
                "lineHeight": 1.75,
              },
              "subtitle2": Object {
                "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
                "fontSize": "0.875rem",
                "fontWeight": 400,
                "letterSpacing": "0.00714em",
                "lineHeight": 1.57,
              },
              "suppressDeprecationWarnings": true,
              "useNextVariants": true,
            },
            "zIndex": Object {
              "appBar": 1100,
              "drawer": 1200,
              "mobileStepper": 1000,
              "modal": 1300,
              "snackbar": 1400,
              "speedDial": 1050,
              "tooltip": 1500,
            },
            Symbol(mui.nested): false,
          }
        }
      >
        <MuiPickersUtilsProvider
          utils={[Function]}
        >
          <Connect(FileUploadRowStatus)
            isUploadInProgress={false}
            onDelete={[MockFunction]}
            progress={0}
          >
            <FileUploadRowStatus
              dispatch={[MockFunction]}
              isUploadInProgress={false}
              locale={
                Object {
                  "deleteHint": "Remove this file",
                  "uploadInProgressText": "Uploading...",
                }
              }
              onDelete={[MockFunction]}
              progress={0}
            >
              <WithStyles(ForwardRef(Tooltip))
                title="Remove this file"
              >
                <ForwardRef(Tooltip)
                  classes={
                    Object {
                      "arrow": "MuiTooltip-arrow",
                      "popper": "MuiTooltip-popper",
                      "popperArrow": "MuiTooltip-popperArrow",
                      "popperInteractive": "MuiTooltip-popperInteractive",
                      "tooltip": "MuiTooltip-tooltip",
                      "tooltipArrow": "MuiTooltip-tooltipArrow",
                      "tooltipPlacementBottom": "MuiTooltip-tooltipPlacementBottom",
                      "tooltipPlacementLeft": "MuiTooltip-tooltipPlacementLeft",
                      "tooltipPlacementRight": "MuiTooltip-tooltipPlacementRight",
                      "tooltipPlacementTop": "MuiTooltip-tooltipPlacementTop",
                      "touch": "MuiTooltip-touch",
                    }
                  }
                  title="Remove this file"
                >
                  <span
                    aria-describedby={null}
                    className=""
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onMouseLeave={[Function]}
                    onMouseOver={[Function]}
                    onTouchEnd={[Function]}
                    onTouchStart={[Function]}
                    title="Remove this file"
                  >
                    <WithStyles(ForwardRef(IconButton))
                      data-testid="undefined-delete"
                      onClick={[MockFunction]}
                    >
                      <ForwardRef(IconButton)
                        classes={
                          Object {
                            "colorInherit": "MuiIconButton-colorInherit",
                            "colorPrimary": "MuiIconButton-colorPrimary",
                            "colorSecondary": "MuiIconButton-colorSecondary",
                            "disabled": "Mui-disabled",
                            "edgeEnd": "MuiIconButton-edgeEnd",
                            "edgeStart": "MuiIconButton-edgeStart",
                            "label": "MuiIconButton-label",
                            "root": "MuiIconButton-root",
                            "sizeSmall": "MuiIconButton-sizeSmall",
                          }
                        }
                        data-testid="undefined-delete"
                        onClick={[MockFunction]}
                      >
                        <WithStyles(ForwardRef(ButtonBase))
                          centerRipple={true}
                          className="MuiIconButton-root"
                          data-testid="undefined-delete"
                          disabled={false}
                          focusRipple={true}
                          onClick={[MockFunction]}
                        >
                          <ForwardRef(ButtonBase)
                            centerRipple={true}
                            className="MuiIconButton-root"
                            classes={
                              Object {
                                "disabled": "Mui-disabled",
                                "focusVisible": "Mui-focusVisible",
                                "root": "MuiButtonBase-root",
                              }
                            }
                            data-testid="undefined-delete"
                            disabled={false}
                            focusRipple={true}
                            onClick={[MockFunction]}
                          >
                            <button
                              className="MuiButtonBase-root MuiIconButton-root"
                              data-testid="undefined-delete"
                              disabled={false}
                              onBlur={[Function]}
                              onClick={[MockFunction]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              tabIndex={0}
                              type="button"
                            >
                              <span
                                className="MuiIconButton-label"
                              >
                                <DeleteIcon>
                                  <WithStyles(ForwardRef(SvgIcon))>
                                    <ForwardRef(SvgIcon)
                                      classes={
                                        Object {
                                          "colorAction": "MuiSvgIcon-colorAction",
                                          "colorDisabled": "MuiSvgIcon-colorDisabled",
                                          "colorError": "MuiSvgIcon-colorError",
                                          "colorPrimary": "MuiSvgIcon-colorPrimary",
                                          "colorSecondary": "MuiSvgIcon-colorSecondary",
                                          "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                          "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                          "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                          "root": "MuiSvgIcon-root",
                                        }
                                      }
                                    >
                                      <svg
                                        aria-hidden="true"
                                        className="MuiSvgIcon-root"
                                        focusable="false"
                                        viewBox="0 0 24 24"
                                      >
                                        <path
                                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                                        />
                                      </svg>
                                    </ForwardRef(SvgIcon)>
                                  </WithStyles(ForwardRef(SvgIcon))>
                                </DeleteIcon>
                              </span>
                              <WithStyles(memo)
                                center={true}
                              >
                                <ForwardRef(TouchRipple)
                                  center={true}
                                  classes={
                                    Object {
                                      "child": "MuiTouchRipple-child",
                                      "childLeaving": "MuiTouchRipple-childLeaving",
                                      "childPulsate": "MuiTouchRipple-childPulsate",
                                      "ripple": "MuiTouchRipple-ripple",
                                      "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                      "rippleVisible": "MuiTouchRipple-rippleVisible",
                                      "root": "MuiTouchRipple-root",
                                    }
                                  }
                                >
                                  <span
                                    className="MuiTouchRipple-root"
                                  >
                                    <TransitionGroup
                                      childFactory={[Function]}
                                      component={null}
                                      exit={true}
                                    />
                                  </span>
                                </ForwardRef(TouchRipple)>
                              </WithStyles(memo)>
                            </button>
                          </ForwardRef(ButtonBase)>
                        </WithStyles(ForwardRef(ButtonBase))>
                      </ForwardRef(IconButton)>
                    </WithStyles(ForwardRef(IconButton))>
                  </span>
                  <ForwardRef(Popper)
                    anchorEl={
                      <span
                        class=""
                        title="Remove this file"
                      >
                        <button
                          class="MuiButtonBase-root MuiIconButton-root"
                          data-testid="undefined-delete"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiIconButton-label"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                              />
                            </svg>
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </span>
                    }
                    className="MuiTooltip-popper"
                    id={null}
                    open={false}
                    placement="bottom"
                    popperOptions={
                      Object {
                        "modifiers": Object {
                          "arrow": Object {
                            "element": null,
                            "enabled": false,
                          },
                        },
                      }
                    }
                    transition={true}
                  />
                </ForwardRef(Tooltip)>
              </WithStyles(ForwardRef(Tooltip))>
            </FileUploadRowStatus>
          </Connect(FileUploadRowStatus)>
        </MuiPickersUtilsProvider>
      </ThemeProvider>
    </Router>
  </MemoryRouter>
</Provider>
`;

exports[`Component FileUploadRowStatus should render circular progress if upload is in progress 1`] = `
<Fragment>
  <WithStyles(ForwardRef(CircularProgress))
    size={20}
    thickness={4}
    value={50}
    variant="determinate"
  />
  <WithStyles(ForwardRef(Typography))
    aria-label="50%"
    variant="caption"
  >
    50%
  </WithStyles(ForwardRef(Typography))>
</Fragment>
`;

exports[`Component FileUploadRowStatus should render default view 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Tooltip))
    title="Remove this file"
  >
    <span>
      <WithStyles(ForwardRef(IconButton))
        data-testid="undefined-delete"
        onClick={[MockFunction]}
      >
        <DeleteIcon />
      </WithStyles(ForwardRef(IconButton))>
    </span>
  </WithStyles(ForwardRef(Tooltip))>
</Fragment>
`;

exports[`Component FileUploadRowStatus should render disabled view 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Tooltip))
    title="Remove this file"
  >
    <span>
      <WithStyles(ForwardRef(IconButton))
        data-testid="undefined-delete"
        disabled={true}
        onClick={[MockFunction]}
      >
        <DeleteIcon />
      </WithStyles(ForwardRef(IconButton))>
    </span>
  </WithStyles(ForwardRef(Tooltip))>
</Fragment>
`;

exports[`Component FileUploadRowStatus should render done icon if upload is finished 1`] = `
<Fragment>
  <WithStyles(ForwardRef(IconButton))>
    <DoneIcon />
  </WithStyles(ForwardRef(IconButton))>
</Fragment>
`;

exports[`Component FileUploadRowStatus should render for edge browser if file is being uploaded but no progress data 1`] = `
<Fragment>
  <WithStyles(ForwardRef(CircularProgress))
    size={20}
    thickness={4}
  />
  <WithStyles(ForwardRef(Typography))
    aria-label="Uploading..."
    variant="caption"
  >
    Uploading...
  </WithStyles(ForwardRef(Typography))>
</Fragment>
`;

exports[`Component FileUploadRowStatus should render if file uploaded successfully but later other file failed 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Tooltip))
    title="Remove this file"
  >
    <span>
      <WithStyles(ForwardRef(IconButton))
        data-testid="undefined-delete"
        onClick={[MockFunction]}
      >
        <DeleteIcon />
      </WithStyles(ForwardRef(IconButton))>
    </span>
  </WithStyles(ForwardRef(Tooltip))>
</Fragment>
`;
